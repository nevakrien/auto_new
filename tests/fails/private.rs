mod m{
	use auto_new::new;

	#[derive(new)]
	#[new_visibility(/*private*/)]
	#[no_new_arc]//check this as well
	pub struct Dumby<'a, T>(&'a T);

	impl<T: Clone> Dumby<'_, T>{
		pub fn do_stuff(&self) -> T{
			return self.0.clone()
		}
	}
}


fn main(){
	let _ = m::Dumby::new(&3);
	let _ = m::Dumby::new_arc(&2);
	println!("hello world");
} 