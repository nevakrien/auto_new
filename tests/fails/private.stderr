error[E0624]: associated function `new` is private
  --> tests/fails/private.rs:18:20
   |
4  |     #[derive(new)]
   |              --- private associated function defined here
...
18 |     let _ = m::Dumby::new(&3);
   |                       ^^^ private associated function

error[E0599]: no function or associated item named `new_arc` found for struct `Dumby` in the current scope
  --> tests/fails/private.rs:19:20
   |
7  |     pub struct Dumby<'a, T>(&'a T);
   |     ----------------------- function or associated item `new_arc` not found for this struct
...
19 |     let _ = m::Dumby::new_arc(&2);
   |                       ^^^^^^^ function or associated item not found in `Dumby<'_, _>`
